pkg/util/util.go

```diff
@@ -1144,6 +1150,49 @@ func ValidateFile(filePath string) error {
 	return nil
 }
 
+// GetGitUrlComponentsFromRaw converts a raw GitHub file link to a map of the url components
+func GetGitUrlComponentsFromRaw(rawGitURL string) (map[string]string, error) {
+	err := ValidateURL(rawGitURL)
+	if err != nil {
+		return nil, err
+	}
+	if !strings.Contains(rawGitURL, "raw.githubusercontent.com") {
+		return nil, errors.Errorf("%s is not a raw github url", rawGitURL)
+	}
+
+	u, _ := url.Parse(rawGitURL)
+	urlPath := strings.SplitN(u.Host+u.Path, "/", 5)
+
+	urlComponents := map[string]string{
```

Comment:
may want to add a short comment with an example raw giturl for better understanding

Reply:
added comment